synapse/handlers/federation_event.py

```diff
@@ -866,6 +866,11 @@ async def _process_pulled_event(
                 event.room_id, event_id, str(err)
             )
             return
+        except Exception as exc:
+            await self._store.record_event_failed_pull_attempt(
+                event.room_id, event_id, str(exc)
+            )
+            raise exc
```

Comment:
I don't have a specific error that I am trying to catch here. But seems good to catch whatever exception might be popping out of our processing loops.